test/integration/http_integration.cc

```diff
@@ -626,6 +626,21 @@ void HttpIntegrationTest::testRetry() {
   EXPECT_EQ(512U, response_->body().size());
 }
 
+// Change the default route to be restrictive, and send a request to an alternate route.
+void HttpIntegrationTest::testGrpcRouterNotFound() {
```

Comment:
Could also consider adding a unit test in `router_test`. I don't think it would bring any increased functional or code coverage, but it's the unit test counterpart, would complement existing gRPC tests there and should be simple. Up to you IMHO.

Reply:
The functionality is now in `http/utility.cc`, added a new unit test to `test/common/http/utility_test.cc`.